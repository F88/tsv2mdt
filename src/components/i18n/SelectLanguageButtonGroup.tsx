import { ToggleButton, ToggleButtonGroup, Tooltip } from '@mui/material';
import { useTranslation } from 'react-i18next';
import { useLanguageContext } from '../../contexts/language-context.js';

export function SelectLanguageButtonGroup() {
  const { language, changeLanguage } = useLanguageContext();
  const { t } = useTranslation();

  return (
    <ToggleButtonGroup
      value={language}
      color={'info'}
      exclusive
      sx={
        {
          // padding: 2,
        }
      }
    >
      <Tooltip title={t('languageButton.switchToEnglish')}>
        <ToggleButton
          value="en"
          aria-label={t('languageButton.switchToEnglish')}
          // color="info"
          onClick={() => {
            changeLanguage('en');
          }}
          // disabled={language === 'en'}
          sx={
            {
              // color: '#e2c96e'
            }
          }
        >
          en
        </ToggleButton>
      </Tooltip>

      <Tooltip title={t('languageButton.switchToFrench')}>
        <ToggleButton
          value="fr"
          aria-label={t('languageButton.switchToFrench')}
          // color="info"
          onClick={() => {
            changeLanguage('fr');
          }}
          // disabled={language === 'fr'}
          sx={
            {
              // color: '#e2c96e'
            }
          }
        >
          fr
        </ToggleButton>
      </Tooltip>

      <Tooltip title={t('languageButton.switchToJapanese')}>
        <ToggleButton
          value="ja"
          aria-label={t('languageButton.switchToJapanese')}
          // color="info"
          onClick={() => {
            changeLanguage('ja');
          }}
          // disabled={language === 'ja'}
          sx={
            {
              // color: '#ff6f69',
              // borderColor: '#ffff69',
            }
          }
        >
          ja
        </ToggleButton>
      </Tooltip>
    </ToggleButtonGroup>
  );
}
