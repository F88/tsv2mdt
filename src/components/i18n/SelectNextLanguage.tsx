import LanguageIcon from '@mui/icons-material/Language';
import { IconButton, Tooltip } from '@mui/material';
import { useLanguageContext } from '../../contexts/language-context.js';

export function SelectNextLanguage() {
  const { setNextLanguage, nextLanguage } = useLanguageContext();

  return (
    <Tooltip title={`Switch language to ${nextLanguage()}`}>
      <IconButton
        onClick={setNextLanguage}
        color="inherit"
        // size="large"
        aria-label={`Switch language to ${nextLanguage()}`}
      >
        <LanguageIcon />
      </IconButton>
    </Tooltip>
  );
}
