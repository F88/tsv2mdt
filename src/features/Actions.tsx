import { Box, Button } from '@mui/material';
import { useTranslation } from 'react-i18next';
import ClearAllIcon from '@mui/icons-material/ClearAll';
import AutoAwesomeIcon from '@mui/icons-material/AutoAwesome';
interface Props {
  onLoadSample: () => void;
  onClearAll: () => void;
}

export function Actions({ onLoadSample, onClearAll }: Props) {
  const { t } = useTranslation();
  return (
    <Box display="flex" gap={2} justifyContent="center" sx={{ mb: 4 }}>
      <Button
        variant="contained"
        onClick={onLoadSample}
        color="primary"
        size="large"
        // color="warning"
        // color="secondary"
        // color="info"
        startIcon={<AutoAwesomeIcon />}
      >
        {t('loadSample')}
      </Button>
      <Button
        variant="outlined"
        size="large"
        onClick={onClearAll}
        // color="secondary"
        startIcon={<ClearAllIcon />}
        color="warning"
      >
        {t('clearAll')}
      </Button>
    </Box>
  );
}
