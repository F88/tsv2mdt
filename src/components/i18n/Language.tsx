import { Stack } from '@mui/material';
import { type ComponentProps } from 'react';
import { SelectLanguageButtonGroup } from './SelectLanguageButtonGroup.jsx';
import { SelectNextLanguage } from './SelectNextLanguage.jsx';

type Props = Pick<ComponentProps<typeof Stack>, 'direction'>;

export function Language(props: Props) {
  return (
    <>
      <Stack direction={props.direction} spacing={0}>
        <SelectNextLanguage />
        <SelectLanguageButtonGroup />
      </Stack>
    </>
  );
}
