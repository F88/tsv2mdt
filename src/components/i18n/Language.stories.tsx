import type { Meta, StoryObj } from '@storybook/react-vite';
import { LanguageContextProvider } from '../../contexts/LanguageContextProvider';
// import { ThemeContextProvider } from '../../contexts/ThemeContextProvider';
// import { MuiThemeProvider } from '../../contexts/MuiThemeProvider';
import { Language } from './Language';
import { LanguageContext } from '../../contexts/language-context';

const meta = {
  title: 'Language',
  component: Language,
  parameters: {
    layout: 'padded',
    // layout: 'fullscreen',
    // layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Language>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Horizontal: Story = {
  args: {
    direction: 'row',
  },
  decorators: (Story) => (
    <LanguageContext
      value={{
        language: 'fr',
        nextLanguage: () => 'ja',
        changeLanguage: (language: string) => {
          console.log(`Language changed to ${language}`);
        },
        setNextLanguage() {
          console.log('Next language set');
        },
      }}
    >
      {/* <ThemeContextProvider> */}
      {/* <MuiThemeProvider> */}
      <Story />
      {/* </MuiThemeProvider> */}
      {/* </ThemeContextProvider> */}
    </LanguageContext>
  ),
};

export const Vertical: Story = {
  args: {
    direction: 'column',
  },
  decorators: (Story) => (
    <LanguageContextProvider>
      {/* <ThemeContextProvider> */}
      {/* <MuiThemeProvider> */}
      <Story />
      {/* </MuiThemeProvider> */}
      {/* </ThemeContextProvider> */}
    </LanguageContextProvider>
  ),
};
