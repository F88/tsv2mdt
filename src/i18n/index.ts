import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import en from './locales/en.json';
import ja from './locales/ja.json';
import fr from './locales/fr.json';

const resources = {
  en: {
    translation: en,
  },
  fr: {
    translation: fr,
  },
  ja: {
    translation: ja,
  },
};

i18n
  .use(initReactI18next)
  .init({
    resources,
    lng: 'en', // default language
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false,
    },
  })
  .catch((error: unknown) => {
    let errorMessage = 'Failed to initialize i18n';
    if (error instanceof Error && error.message) {
      errorMessage += `: ${error.message}`;
    }
    console.error(errorMessage, error);
  });

export default i18n;
