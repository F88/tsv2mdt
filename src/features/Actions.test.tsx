import { cleanup, render, screen } from '@testing-library/react';
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
import { I18nextProvider } from 'react-i18next';
import { Actions } from './Actions';
import i18n from '../../test/helpers/i18n-for-tests';
import userEvent from '@testing-library/user-event';

describe('Actions', () => {
  const mockOnLoadSample = vi.fn();
  const mockOnClearAll = vi.fn();

  const renderActions = () => {
    return render(
      <I18nextProvider i18n={i18n}>
        <Actions onLoadSample={mockOnLoadSample} onClearAll={mockOnClearAll} />
      </I18nextProvider>,
    );
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  afterEach(() => {
    cleanup();
  });

  it('renders load sample button with correct text and icon', () => {
    renderActions();

    const loadSampleButton = screen.getByRole('button', {
      name: /Load Sample/i,
    });
    expect(loadSampleButton).toBeInTheDocument();
    expect(loadSampleButton).toHaveAttribute('type', 'button');

    // snapshot testing
    expect(loadSampleButton).toMatchSnapshot();
  });

  it('renders clear all button with correct text and icon', () => {
    renderActions();

    const clearAllButton = screen.getByRole('button', { name: /Clear All/i });
    expect(clearAllButton).toBeInTheDocument();
    expect(clearAllButton).toHaveAttribute('type', 'button');

    // snapshot testing
    expect(clearAllButton).toMatchSnapshot();
  });

  it('calls onLoadSample when load sample button is clicked', async () => {
    const user = userEvent.setup();
    renderActions();

    const loadSampleButton = screen.getByRole('button', {
      name: /Load Sample/i,
    });
    await user.click(loadSampleButton);

    expect(mockOnLoadSample).toHaveBeenCalledTimes(1);
  });

  it('calls onClearAll when clear all button is clicked', async () => {
    const user = userEvent.setup();
    renderActions();

    const clearAllButton = screen.getByRole('button', { name: /Clear All/i });
    await user.click(clearAllButton);

    expect(mockOnClearAll).toHaveBeenCalledTimes(1);
  });

  it('renders buttons with correct styling properties', () => {
    renderActions();

    const loadSampleButton = screen.getByRole('button', {
      name: /load sample/i,
    });
    const clearAllButton = screen.getByRole('button', { name: /Clear All/i });

    expect(loadSampleButton).toHaveClass('MuiButton-contained');
    expect(clearAllButton).toHaveClass('MuiButton-outlined');

    // snapshot testing
    expect(loadSampleButton).toMatchSnapshot();
    expect(clearAllButton).toMatchSnapshot();
  });
});
