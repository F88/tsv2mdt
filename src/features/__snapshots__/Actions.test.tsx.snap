// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Actions > renders buttons with correct styling properties 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-79zo2g-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeLarge css-hev6se-MuiButton-startIcon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
      data-testid="AutoAwesomeIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m19 9 1.25-2.75L23 5l-2.75-1.25L19 1l-1.25 2.75L15 5l2.75 1.25zm-7.5.5L9 4 6.5 9.5 1 12l5.5 2.5L9 20l2.5-5.5L17 12zM19 15l-1.25 2.75L15 19l2.75 1.25L19 23l1.25-2.75L23 19l-2.75-1.25z"
      />
    </svg>
  </span>
  Load Sample
</button>
`;

exports[`Actions > renders buttons with correct styling properties 2`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedWarning MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorWarning MuiButton-root MuiButton-outlined MuiButton-outlinedWarning MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorWarning css-15borzz-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeLarge css-hev6se-MuiButton-startIcon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
      data-testid="ClearAllIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M5 13h14v-2H5zm-2 4h14v-2H3zM7 7v2h14V7z"
      />
    </svg>
  </span>
  Clear All
</button>
`;

exports[`Actions > renders clear all button with correct text and icon 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedWarning MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorWarning MuiButton-root MuiButton-outlined MuiButton-outlinedWarning MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorWarning css-15borzz-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeLarge css-hev6se-MuiButton-startIcon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
      data-testid="ClearAllIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M5 13h14v-2H5zm-2 4h14v-2H3zM7 7v2h14V7z"
      />
    </svg>
  </span>
  Clear All
</button>
`;

exports[`Actions > renders load sample button with correct text and icon 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-79zo2g-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeLarge css-hev6se-MuiButton-startIcon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
      data-testid="AutoAwesomeIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m19 9 1.25-2.75L23 5l-2.75-1.25L19 1l-1.25 2.75L15 5l2.75 1.25zm-7.5.5L9 4 6.5 9.5 1 12l5.5 2.5L9 20l2.5-5.5L17 12zM19 15l-1.25 2.75L15 19l2.75 1.25L19 23l1.25-2.75L23 19l-2.75-1.25z"
      />
    </svg>
  </span>
  Load Sample
</button>
`;
